tests/NationalInstruments.Analyzers.UnitTests/LongChainDottedInvocationAnalyzerTests.cs

```diff
@@ -0,0 +1,331 @@
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a method
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        Test(IEnumerable<ISoftwareContent> selectedSoftwareContents)
+        {
+            var distinctSelectedSoftware = selectedSoftwareContents
+                .GroupBy(software => software.AliasName)
+                .Select(g => g.First())
+                .Distinct();
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a constructor
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        private IEnumerable<ISoftwareContent> selectedSoftwareContents = Enumerable.Empty<ISoftwareContent>();
+
+        IEnumerable<ISoftwareContent> DistinctSoftwareContents =>
+         selectedSoftwareContents
+            .GroupBy(software => software.AliasName)
+            .Select(g => g.First())
+            .Distinct();
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+"
+                }, // inside a property getter
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> softwareContent)
+        {
+            softwareContent.Select(content => content
+                .Children.GroupBy(software => software.AliasName)
+                     .Select(g => g.First())
+                     .Distinct());
+        }
+
+        private interface ISoftwareContent
+        {
+            IEnumerable<ISoftwareContent> Children { get; }
+
+            string AliasName { get; }
+        }
+    }
+}
+"
+                } // inside an argument
+            };
+
+        public static IEnumerable<object[]> PoorlySplitStatementInVariousContextsWithDiagnosticLocation =>
+            new[]
+            {
+                new object[]
+                {
+                    @"
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NationalInstruments.Analyzers.UnitTests
+{
+    class Test
+    {
+        void Foo(IEnumerable<ISoftwareContent> selectedSoftwareContents)
+        {
+            var distinctSelectedSoftware = selectedSoftwareContents.GroupBy(software => software.AliasName).Select(g => g.First()).Distinct();
+        }
+
+        private interface ISoftwareContent
+        {
+            string AliasName { get; }
+        }
+    }
+}
+", (11, 44)
```

Comment:
Instead of statically coding this in (which is difficult to determine...especially if the code is dynamically created), you can use the special markers <?>, <|...|>, and <|> where the error should occur. Then, instead of `TestFile`, you use `AutoTestFile` and supply it rules. If you don't want to wait for my follow-up comment, search the existing unit tests for `AutoTestFile` to see usage.

Reply:
Thanks, I did not know about `AutoTestFile`! I will update the tests.